### DISCUSSION TITLE
Noisy audio after adding early training vocoder checkpoint.

### DISCUSSION THREAD

**Discussion by Sadam1195** [coqui-ai/TTS#464] [Show and tell]
I tacotron with new data (male voice **2**) using previously trained checkpoint (female voice **1**) . 
After around 90k steps I stopped tacotron training and started training wavernn (on the data (male voice **3**) which is neither from 1 and 2) .

I trained wavernn for around 45k steps (still in training) and used a trained checkpoint of 45000 steps of wavernn training model and tried to see the difference but results were quite different than I expected. With vocoder supplied the generated voice sample was very noisy as compared to the that of without vocoder. 

Why is that? Is that because of wavernn is trained on different data then the taco or it is because of the early training checkpoint? Wavernn model is decreasing and currently around 5.9

```
from TTS.utils.synthesizer import Synthesizer

# run your own TTS model (Using Griffin-Lim Vocoder)
# ./TTS/bin/synthesize.py --text "Text for TTS" --model_path path/to/model.pth.tar --config_path path/to/config.json --out_path output/path/speech.wav

# run your own TTS and Vocoder models
# ./TTS/bin/synthesize.py --text "Text for TTS" --model_path path/to/config.json --config_path path/to/model.pth.tar --out_path output/path/speech.wav
    # --vocoder_path path/to/vocoder.pth.tar --vocoder_config_path path/to/vocoder_config.json

vocoder_path = "wav_models_dir/wavernn_ljspeech-April-29-2021_08+32PM-0000000/checkpoint_47500.pth.tar"
# vocoder_config_path = "wav_models_dir/wavernn_ljspeech-April-29-2021_08+32PM-0000000/config.json"
vocoder_config_path = "2wavernn_config.json"
model_path = "D_Models/LJSpeech/ljspeech-ddc-April-28-2021_10+22AM-0000000/checkpoint_470000.pth.tar"
config_path = "2d_config.json"
out_path = "output3.wav"

text = "En sendas sentencias de 2013, esa Sala del Tribunal Supremo anuló dos indultos porque no estaban suficientemente motivados."

# RUN THE SYNTHESIS
# load models
synthesizer = Synthesizer(model_path, config_path, vocoder_path, vocoder_config_path, use_cuda=True)

print(" > Text: {}".format(text))


# kick it
wav = synthesizer.tts(text)

# save the results
print(" > Saving output to {}".format(out_path))
synthesizer.save_wav(wav, out_path)
```

**Output**

```
 > Using model: Tacotron2
 > Generator Model: WaveRNN
 > Text: En sendas sentencias de 2013, esa Sala del Tribunal Supremo anuló dos indultos porque no estaban suficientemente motivados.
 > Text splitted to sentences.
['En sendas sentencias de 2013, esa Sala del Tribunal Supremo anuló dos indultos porque no estaban suficientemente motivados.']
150000/150016 -- batch_size: 1 -- gen_rate: 0.8 kHz -- x_realtime: 0.0   > Processing time: 190.68889927864075
 > Real-time factor: 19.097536232212395
 > Saving output to output3.wav

```

Also why is `batch_size: 1` always 1 when synthesizing? 

My audio parameters are match in both configs

https://user-images.githubusercontent.com/42005385/116673557-fa1bbe00-a9bc-11eb-9b72-d5e18569aff2.mp4

Early checkpoint

https://user-images.githubusercontent.com/42005385/116673572-fe47db80-a9bc-11eb-80d1-597c402e8b0c.mp4

Updated checkpoint 

https://user-images.githubusercontent.com/42005385/116673598-03a52600-a9bd-11eb-9d99-35e23ba1d6ab.mp4

**Comment by Sadam1195** [comment]
Indeed, it was the due to early training checkpoint. Glad no one responded as I continued training, otherwise I would have halted my training and would have tried something different. 
> Why is that? Is that because of wavernn is trained on different data then the taco or **it is because of the early training checkpoint**? Wavernn model is decreasing and currently around 5.9


This is the latest output. `model loss 5.32`

https://user-images.githubusercontent.com/42005385/116870149-67cd2180-ac2b-11eb-8f1c-de41dee802a8.mp4

**Reply by erogol** [reply]
vocoders need more training steps to generate good results. You should not complain until the model training converges.

**Reply by Sadam1195** [reply]
> vocoders need more training steps to generate good results. You should not complain until the model training converges.

Not complaining but it's exhausting though.
